test/unit/math/mix/functor/root_finder_test.cpp

```diff
@@ -0,0 +1,130 @@
+  std::frexp(x, &exponent);
+  // Rough guess is to divide the exponent by three.
+  double guess = ldexp(1., exponent / 3);
+  // Minimum possible value is half our guess.
+  double min = ldexp(0.5, exponent / 3);
+  // Maximum possible value is twice our guess.
+  double max = ldexp(2., exponent / 3);
+  // Maximum possible binary digits accuracy for type T.
+  const int digits = std::numeric_limits<double>::digits;
+  int get_digits = static_cast<int>(digits * 0.4);
+  std::uintmax_t maxit = 20;
+  auto f = [](const auto& g, const auto& x) { return g * g * g - x; };
+  auto f_g = [](const auto& g, const auto& x) { return 3 * g * g; };
+  auto f_gg = [](const auto& g, const auto& x) { return 6 * g; };
+  auto full_f = [&f, &f_g, &f_gg, guess, min, max](auto&& xx) {
+    return root_finder(std::make_tuple(f, f_g, f_gg), guess, min, max, xx);
+  };
+  stan::test::expect_ad(full_f, x);
+}
+
+TEST(MixFun, root_finder_fifth) {
+  using stan::math::root_finder;
+  // return cube root of x using 1st and 2nd derivatives and Halley.
+  // using namespace std;  // Help ADL of std functions.
+  double x = 27;
+  int exponent;
+  // Get exponent of z (ignore mantissa).
+  std::frexp(x, &exponent);
+  // Rough guess is to divide the exponent by three.
+  double guess = ldexp(1., exponent / 5);
+  // Minimum possible value is half our guess.
+  double min = ldexp(0.5, exponent / 5);
+  // Maximum possible value is twice our guess.
+  double max = ldexp(2., exponent / 5);
+  // Maximum possible binary digits accuracy for type T.
+  const int digits = std::numeric_limits<double>::digits;
+  int get_digits = static_cast<int>(digits * 0.4);
+  std::uintmax_t maxit = 20;
+  auto f = [](const auto& g, const auto& x) { return g * g * g * g * g - x; };
+  auto f_g = [](const auto& g, const auto& x) { return 5 * g * g * g * g; };
+  auto f_gg = [](const auto& g, const auto& x) { return 20 * g * g * g; };
+  auto full_f = [&f, &f_g, &f_gg, guess, min, max](auto&& xx) {
+    return root_finder(std::make_tuple(f, f_g, f_gg), guess, min, max, xx);
+  };
+  stan::test::expect_ad(full_f, x);
+}
+*/
+TEST(MixFun, root_finder_beta_log) {
+  auto beta_lpdf_deriv = [](auto&& x, auto&& alpha, auto&& beta, auto&& p) {
+    using stan::math::log;
+    using stan::math::lgamma;
+    using stan::math::log1m;
+    auto val = -((beta * log1m(x) + (-2 + alpha) * log(x)
+                  + log(1 - 2 * x - alpha + x * alpha + x * beta)
+                  + lgamma(alpha + beta))
+                 - (2 * log1m(x) + lgamma(alpha) + lgamma(beta)));
+    return val;
+  };
+  auto f = [](auto&& x, auto&& alpha, auto&& beta, auto&& p) {
+    return stan::math::beta_lcdf(x, alpha, beta) - p;
+  };
+  auto f_g = [](auto&& x, auto&& alpha, auto&& beta, auto&& p) {
+    return stan::math::beta_lpdf(x, alpha, beta);
+  };
+  auto f_gg
+      = [&beta_lpdf_deriv](auto&& x, auto&& alpha, auto&& beta, auto&& p) {
+          return beta_lpdf_deriv(x, alpha, beta, p);
+        };
+  double guess = .3;
```

Comment:
Why .3? Does the test pass with other initial guesses?

Reply:
.3 is just a wild guess from me. In the test below I use .5 as well. @bgoodri should there be some way we cleverly set the initial guess for the beta test?